source/shared/cpp/ObjectModel/MarkDownBlockParser.cpp

```diff
@@ -655,7 +676,7 @@ void OrderedListParser::Match(std::stringstream& stream)
 {
     // used to capture digit char
     std::string number_string = "";
-    if (isdigit(stream.peek()))
+    if (stream.peek() > 0 && isdigit(stream.peek()))
```

Comment:
>stream.peek() > 0 [](start = 8, length = 17)

Do you think it might make this easier to read if we put the 0 comparison check behind a helpfully named inline function? (IsExtendedCharacter or something?) There's a lot of places where we use this. We could even go so far as to have our own is digit function (and others) that includes that check so in the logic we're just checking for digit and we don't have the cognitive overhead about thinking about this issue at every check.

Reply:
I was a bit on the fence about it, but I'll take another look. One complicating factor is EOF, which is `0xFF`... Is that an extended character? I didn't know how to answer, so I ended up doing nothing.

---
In reply to: [314840442](https://github.com/microsoft/AdaptiveCards/pull/3342#discussion_r314840442) [](ancestors = 314840442)